src/gdl2gdl.cpp

```diff
@@ -103,13 +104,27 @@ static void g2gPerformCallbackProcedure(pid_t pid) {
 }
 
 void g2gAsynchronousReturnTrap() {
-  while (g2gMap.size() > 0) {
-    for (g2gMapIter it = g2gMap.begin(); it != g2gMap.end(); ++it) {
-	  if ( (*it).second.status==1 && (*it).second.nowait ) gdl_ipc_read_client_nowait((*it).first);
-	}
-	usleep (10000);
+  while (!g2gMap.empty()) {
+    for (auto g2gMapIter = g2gMap.cbegin(); g2gMapIter != g2gMap.cend();) {
+      if ((*g2gMapIter).second.status == 1 && (*g2gMapIter).second.nowait) {
+        gdl_ipc_read_client_nowait((*g2gMapIter).first);
+      }
+      // we need to take care of our children, use a no-hang wait call to check the pid
+      int status = -1;
+      if (waitpid((*g2gMapIter).first, &status, WUNTRACED | WCONTINUED | WNOHANG) == -1) {
+        // we're out of luck, wait returns an error, so break the loop and exit
+        std::cerr << "g2gAsynchronousReturnTrap exiting" << std::endl;
+        break;
+      }
+      // remove stopped children from the map
+      if (WIFEXITED(status) || WIFSIGNALED(status))
+        g2gMap.erase(g2gMapIter++);
+        // we need to manually count here, since we're modifying the map in the loop
+      else
+        ++g2gMapIter;
```

Comment:
It seems the previous implementation expected some process (the child?) to take care of changing this map. If this is the case, this code introduces an error. It only works as intended, if the map is filled properly before the thread executing this function is started and the thread executing this code is the 
*only* one modifying it. Otherwise we would need a mutex and probably some more changes.

Reply:
@jkohnert thanks for you efforts. It seems this version cannot run IDL's "idlbridge_demo" procedure anymore. Will have a look, but would not a quick patch be  to temporarily set SIGCHILD to something before the spawn and ressetting it to SIG_IGN just after? Chances that IDL_IDLBridge and SPAWN are run simultaneously are minimal. A more robust solution can be presented afterwards.